spopt/locate/coverage.py

```diff
@@ -314,6 +313,364 @@ def solve(self, solver: pulp.LpSolver, results: bool = True):
+                        )
+        else:
+            raise AttributeError(
+                "before setting constraints must set facility variable"
+            )
+
+    @classmethod
+    def from_cost_matrix(
+        cls,
+        cost_matrix: np.array,
+        service_radius: float,
+        solver: pulp.LpSolver, 
+        predefined_facilities_arr: np.array = None,
+        name: str = "LSCP-B",
+        
+    ):
+        """
+        Create a LSCPB object based on a cost matrix.
+
+        Parameters
+        ----------
+        cost_matrix: np.array
+            two-dimensional distance array between facility points and demand point
+        service_radius: float
+            maximum acceptable service distance by problem
+        name: str, default="LSCP-B"
+            name of the problem
+
+        Returns
+        -------
+        LSCPB object
+
+        Examples
+        --------
+        >>> from spopt.locate.coverage import LSCPB, LSCP
+        >>> from spopt.locate.util import simulated_geo_points
+        >>> import pulp
+        >>> import spaghetti
+
+        Create regular lattice
+
+        >>> lattice = spaghetti.regular_lattice((0, 0, 10, 10), 9, exterior=True)
+        >>> ntw = spaghetti.Network(in_data=lattice)
+        >>> street = spaghetti.element_as_gdf(ntw, arcs=True)
+        >>> street_buffered = geopandas.GeoDataFrame(
+        ...                            geopandas.GeoSeries(street["geometry"].buffer(0.2).unary_union),
+        ...                            crs=street.crs,
+        ...                            columns=["geometry"])
+
+        Simulate points belong to lattice
+
+        >>> demand_points = simulated_geo_points(street_buffered, needed=100, seed=5)
+        >>> facility_points = simulated_geo_points(street_buffered, needed=5, seed=6)
+
+        Snap points to the network
+
+        >>> ntw.snapobservations(demand_points, "clients", attribute=True)
+        >>> clients_snapped = spaghetti.element_as_gdf(ntw, pp_name="clients", snapped=True)
+        >>> ntw.snapobservations(facility_points, "facilities", attribute=True)
+        >>> facilities_snapped = spaghetti.element_as_gdf(ntw, pp_name="facilities", snapped=True)
+
+        Calculate the cost matrix
+        >>> cost_matrix = ntw.allneighbordistances(
+        ...    sourcepattern=ntw.pointpatterns["clients"],
+        ...    destpattern=ntw.pointpatterns["facilities"])
+
+        Create LSCPB instance from cost matrix
+
+        >>> lscpb_from_cost_matrix = LSCPB.from_cost_matrix(cost_matrix, max_coverage=8, pulp.PULP_CBC_CMD(msg=False)
+        >>> lscpb_from_cost_matrix = lscpb_from_cost_matrix.solve(pulp.PULP_CBC_CMD(msg=False))
+
+        Get facility lookup demand coverage array
+
+        >>> lscpb_from_cost_matrix.fac2cli
+
+        """
+        # create a lscp object
+        lscp = LSCP.from_cost_matrix(cost_matrix, service_radius)
+
+        #solve lscp
+        lscp.solve(solver)
+
```

Comment:
It might be useful to store the LSCP objective value as a class attribute. Then it can be use later in the model *and* kept for info/diagnostics later. Something like the following or thereabouts:

```python
self.lscp_obj_value = lscp.problem.objective.value()
```

Reply:
@jGaboardi I updated the LSCPB class to include this attribute. I also updated the from_cost_matrix class method to set this attribute. However, it seems my implementation isn't quite in line with the package style and I'd like some feedback. Thanks!

Reply:
hmmm. I see what you mean. Maybe we can (1) declare the `lscp_obj_value` variable within `from_cost_matrix`; (2) pass it into `LSCPB`; assign it as an attribute to the new 


I think `lscp_obj_value` does not need to be set within `LSCPB`, but can be set directly after instantiation. This is because it is not used until we are adding the facility constraint. For example:

```python
lscpb = LSCPB(name, model)
lscpb.lscp_obj_value = lscp.problem.objective.value()

...

FacilityModelBuilder.add_facility_constraint(
    lscpb, lscpb.problem, lscpb.lscp_obj_value
)
```